src/Messages/Attachments/TSAttachmentStream.m

```diff
@@ -112,30 +116,110 @@ + (NSUInteger)numberOfItemsInAttachmentsFolder
     return count;
 }
 
-- (nullable NSString *)filePath
+- (nullable NSString *)buildLocalFilePath
 {
-    return [MIMETypeUtil filePathForAttachment:self.uniqueId
-                                    ofMIMEType:self.contentType
-                                      filename:self.filename
-                                      inFolder:[[self class] attachmentsFolder]];
+    if (!self.localRelativeFilePath) {
+        return nil;
+    }
+
+    return [[[self class] attachmentsFolder] stringByAppendingPathComponent:self.localRelativeFilePath];
+}
+
+- (nullable NSString *)localFilePathWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+{
```

Comment:
Yes!  This approach is horrible... but I couldn't think of something better.

> Would it simplify things to create the localfile path in the corresponding init methods?

You're right. I was trying to avoid this since we don't want to create a transaction within a transaction, but this can be avoided by doing the write asynchronously.